packages/arb-token-bridge-ui/src/state/cctpState.ts

```diff
@@ -500,13 +480,73 @@ export function useCctpFetching({
     }
   }, [withdrawals, setTransfers])
 
+  const setPendingTransfer = useCallback(
+    (transfer: PartialMergedTransaction, isDeposit: boolean) => {
+      const mutate = isDeposit ? mutateDeposits : mutateWithdrawals
+      mutate(
+        {
+          pending: [transfer as MergedTransaction],
+          completed: []
+        },
+        {
+          populateCache(result, currentData) {
+            if (!currentData) {
+              return result
+            }
+            const index = currentData.pending.findIndex(
+              tx => tx.txId === result.pending[0].txId
+            )
+            const previousData = { ...currentData } // Make sure we don't mutate previous data
+            const existingTransfer = previousData.pending[index]
+            if (existingTransfer) {
+              previousData.pending[index] = {
+                ...existingTransfer,
+                ...result.pending[0],
+                cctpData: {
+                  ...existingTransfer.cctpData,
+                  ...result.pending[0].cctpData
+                }
+              }
+              return previousData
+            }
+
+            return {
+              pending: result.pending.concat(currentData.pending),
```

Comment:
nothing against `concat` but i think `[...result.pending, currentData.pending]` would reduce cognitive load when reading it